Squadrosu.Framework/Piece.cs

```diff
@@ -26,13 +26,18 @@ public class Piece
     /// Line number
     /// </summary>
     public readonly int LineNumber;
+    /// <summary>
+    /// The board in which the piece plays
+    /// </summary>
+    public readonly Board Board;
 
-    public Piece(Player Player, int LineNumber)
+    public Piece(Player Player, int LineNumber, Board Board)
     {
         Direction = Direction.Forward;
         Position = 0;
         this.LineNumber = LineNumber;
         this.Player = Player;
+        this.Board = Board;
     }
```

Comment:
C'est marrant que je ne le remarque que maintenant, mais les arguments ne sont pas censés être en `PascalCase`, normalement ton éditeur a dû te le dire
```suggestion
    public Piece(Player player, int lineNumber, Board board)
    {
        Direction = Direction.Forward;
        Position = 0;
        LineNumber = lineNumber;
        Player = player;
        Board = board;
    }
```